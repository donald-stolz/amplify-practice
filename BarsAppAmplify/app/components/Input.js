// @flow
import React from 'react';
import {
  View, TextInput, StyleSheet, Dimensions,
} from 'react-native';

// Types
type Props = {
  onChangeText: Function,
  value: string,
  placeholder: string,
  secureTextEntry?: boolean,
  style?: {},
};

const Input = ({
  onChangeText, value, placeholder, secureTextEntry, style,
}: Props) => (
  <View>
    <TextInput
      placeholder={placeholder}
      returnKeyType="done"
      underlineColorAndroid="transparent"
      style={[styles.inputStyle, style]}
      onChangeText={onChangeText}
      value={value}
      secureTextEntry={secureTextEntry}
      autoCorrect={false}
      autoCapitalize="none"
    />
  </View>
);

const { width } = Dimensions.get('window');

const styles = StyleSheet.create({
  inputStyle: {
    backgroundColor: 'white',
    marginVertical: 10,
    paddingTop: 10,
    paddingBottom: 10,
    paddingLeft: 10,
    paddingRight: 10,
    width: width * 0.8,
    borderRadius: 8,
  },
});

Input.defaultProps = {
  secureTextEntry: false,
  style: {},
};

export default Input;
