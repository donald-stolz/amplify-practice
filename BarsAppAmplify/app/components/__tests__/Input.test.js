import React from 'react';
import renderer from 'react-test-renderer';
import Input from '../Input';

test('Input renders correctly', () => {
  const onChangeText = jest.fn();
  const tree = renderer
    .create(<Input placeholder="test@test.com" onChangeText={onChangeText} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
