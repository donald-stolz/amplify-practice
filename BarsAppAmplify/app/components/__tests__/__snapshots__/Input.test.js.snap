// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<View>
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    autoCorrect={false}
    onChangeText={[MockFunction]}
    placeholder="test@test.com"
    returnKeyType="done"
    secureTextEntry={false}
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderRadius": 8,
          "marginVertical": 10,
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
          "width": 600,
        },
        Object {},
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
